### PR TITLE
Fix T/S advective and horizontal diffusive fluxes and change T units

### PR DESCRIPTION
**Pull request by ashao**
The advection and diffusive diagnostics are all extensive fields and so the
v_extensive flag needs to be set to true for the remapped version of these
fields to be done correctly. Per
https://github.com/NOAA-GFDL/MOM6-examples/issues/167, the units for
temperature have been changed to W/m^2 by including a conversion factor in
the register_diag_field call. If Boussinesq = .false., a warning is thrown
that the diagnostics are converted to Watts using a Boussinesq
approximation.

### CODE REVIEW

**Comment by StephenGriffies** [comment]
Are you sure we cannot do a proper conversion with non-Boussinesq case?  The thickness factors should already have the in situ density incorporated, since thickness is pressure related.

**Comment by StephenGriffies** [comment]
The non-bouss is not a priority for now.  But something to think about.  I believe it should all be there.  Perhaps only a cp_ocean factor is needed rather than rho0*cp_ocean?

File: src/core/MOM.F90 [hunk 1]
```diff
@@ -2339,8 +2341,13 @@ subroutine register_diags(Time, G, GV, CS, ADp)
 
   thickness_units = get_thickness_units(GV)
   flux_units      = get_flux_units(GV)
-  T_flux_units    = get_tr_flux_units(GV, "degC")
-  S_flux_units    = get_tr_flux_units(GV, "psu")
+  T_flux_units    = trim("Watts/m2")
```
Comment (adcroft) [inline src/core/MOM.F90]: CMIP units are "W m-2". I'll change it when merging.

http://clipc-services.ceda.ac.uk/dreq/index/units.html

File: src/core/MOM.F90 [hunk 2]
```diff
@@ -2339,8 +2341,13 @@ subroutine register_diags(Time, G, GV, CS, ADp)
...
+  S_flux_units    = get_tr_flux_units(GV, "PPT")
```
Comment (adcroft) [inline src/core/MOM.F90]: CMIP (and model) units are "psu".

**Comment by adcroft** [comment]
https://gitlab.gfdl.noaa.gov/ogrp/MOM6/pipelines/2544

**Comment by ashao** [comment]
@StephenGriffies I don't know enough about the model when run in a non-Boussinesq configuration to answer confidently. Hopefully the runtime warning will serve as a reminder for our future selves to delve into the problem.

File: src/core/MOM.F90 [hunk 3]
```diff
@@ -2339,8 +2341,13 @@ subroutine register_diags(Time, G, GV, CS, ADp)
...
+  if (.not. GV%Boussinesq) then
+    call MOM_error(WARNING, "register_diags: Conversion from temperature flux to W m2 for advective" // &
```
Comment (adcroft) [inline src/core/MOM.F90]: The conversion for non-Boussinesq is correct. I'll remove this warning in the merge too.

Comment (StephenGriffies) [reply src/core/MOM.F90]: Great. I thought the model was properly setup to do non-Bouss seamlessly.

**Pull request merged**